File: bootcamps/bootcamp_2020/solutions/exercise2_solution.cpp
////////////////////////////////////////////////////////////////////////////////
////////////////////////////////////////////////////////////////////////////////
////	exercise2_solution.cpp
////
////	September 2020,  baclark@msu.edu
////	Example solution for the second exercise in bootcamp 2020
////
////	In this example, I will use ROOT to make the plot
////////////////////////////////////////////////////////////////////////////////

// C/C++ Includes
#include <iostream>

//AraRoot Includes
#include "AraGeomTool.h"

//ROOT Includes
#include "TGraph.h"
#include "TCanvas.h"

using namespace std;

int main(int argc, char **argv)
{

	if(argc<2) {  // Check to make sure there are enough arguments to do something meaningful
		std::cout << "Usage requires you to provide input parameter of the form " << basename(argv[0]) << " <station>" << std::endl;
		return -1;
	}

	int station = atoi(argv[1]); // get the station number

	AraGeomTool *geomTool = AraGeomTool::Instance();

	// the x-y coordinates of channels 0-3 are enough for a top down view
	std::vector<double> x;
	std::vector<double> y;

	for(int i=0; i<4; i++){
		Double_t *ant_location = geomTool->getStationInfo(station)->getAntennaInfo(i)->antLocation;
		x.push_back(ant_location[0]);
		y.push_back(ant_location[1]);
	}

	TGraph *g = new TGraph(4, &x[0], &y[0]);

	TCanvas *c = new TCanvas("c","c", 1100,850);
	g->Draw("AP");
	g->SetMarkerStyle(8);
	g->SetMarkerSize(2);
	g->SetTitle("Borehole locations of A2; X [m]; Y [m]");
	c->SaveAs("ant_locations_cpp.png");

}//close the main program

File: bootcamps/bootcamp_2020/solutions/exercise2_solution.py
############################################################################
############################################################################
####	exercise2_solution.py
####
####	September 2020,  baclark@msu.edu
####	Example solution for the second exercise in bootcamp 2020
####
####	In this example, I will use python to make the plot
############################################################################


import ROOT

import matplotlib.pyplot as plt

import os

ROOT.gSystem.Load(os.environ.get('ARA_UTIL_INSTALL_DIR')+"/lib/libAraEvent.so")

# create a geomtool
geomTool = ROOT.AraGeomTool.Instance()

station = 2

# make storge containers for the x and y coordinates
x = []
y = []

# the x-y coordinates of channels 0-3 are enough for a top down view
for ant in range(3):
	ant_location = geomTool.getStationInfo(2).getAntennaInfo(ant).antLocation
	x.append(ant_location[0])
	y.append(ant_location[1])

# make a plot
fig = plt.figure(figsize=(5,5))
ax = fig.add_subplot(111)
ax.hist(list_of_snrs, bins=np.arange(0,10,0.5))
ax.set_ylabel('X [m]')
ax.set_xlabel('Y [m]')
ax.set_title('Borehole locations of A2')
fig.savefig('ant_locations_py.png', bbox_inches='tight', dpi=300)

